guppylang/checker/linearity_checker.py

```diff
@@ -332,7 +332,7 @@ def _reassign_single_inout_arg(self, place: Place, node: AstNode) -> None:
         # Places involving subscripts are given back by visiting the `__setitem__` call
         if subscript := contains_subscript(place):
             assert subscript.setitem_call is not None
-            self.visit(subscript.setitem_call)
+            self.visit(subscript.setitem_call[0])
```

Comment:
We should assign the temporary to `self.dfg[place]` before compiling the call

Reply:
The visit here is not compilation - how can I assign it when this is inside the linearity checker?